src/components/calcite-tree-item/calcite-tree-item.tsx

```diff
@@ -429,4 +429,11 @@ export class CalciteTreeItem implements ConditionalSlotComponent {
       return;
     }
   };
+
+  private isSelectionMulti(): boolean {
+    return (
+      this.selectionMode === TreeSelectionMode.Multi ||
```

Comment:
To simplify, can you introduce an internal `isSelectionMulti` variable that gets computed whenever `selectionMode` is updated? The latter is an internal prop already, so you should be able to watch for changes.